apiserver/facades/client/storage/storage.go

```diff
@@ -30,6 +33,29 @@ type BlockDeviceService interface {
 	) ([]blockdevice.BlockDevice, error)
 }
 
+// RemovalService defines the interface required for removing storage related
+// entities in the model on behalf of an API caller.
+type RemovalService interface {
+	// RemoveStorageAttachmentsFromAliveUnit is reponsible for removing one or
+	// more storage attachments from a unit that is still alive in the model.
+	// This operation can be considered a detatch of a storage instance from a
+	// unit.
+	//
+	// The following errors may be returned:
+	// - [storageerrors.StorageAttachmentNotFound] if the supplied storage
+	// attachment uuid does not exist in the model.
+	// - [applicationerrors.UnitNotAlive] if the unit the storage attachment is
+	// conencted to is not alive.
+	// [applicationerrors.UnitStorageMinViolation] if removing a storage
+	// attachment would violate the charm minimums required for the unit.
+	RemoveStorageAttachmentFromAliveUnit(
+		context.Context,
+		domainstorageprovisioning.StorageAttachmentUUID,
+		bool,
+		time.Duration,
```

Comment:
These two have no meaning without names yet we have full documentation on the errors an implementation returns?